[Yap Jie Xiang] iP

akgrenSoar opened the pull request:
Please give me as many feedback as possible, very much appreciated!

Commit:
Add Gradle support

akgrenSoar committed:
ChatbotEcho

akgrenSoar committed:
Level-1 (v2)

akgrenSoar committed:
Level-2

akgrenSoar committed:
Level-3

akgrenSoar committed:
Level-4

akgrenSoar committed:
A-TextUiTesting

akgrenSoar committed:
A-TextUiTesting

akgrenSoar committed:
A-TextUiTesting

akgrenSoar committed:
Level-5

akgrenSoar committed:
Level-6

akgrenSoar committed:
A-Enums

akgrenSoar committed:
Rename DukeCmdKey to DukeCmdEnum

akgrenSoar committed:
Fix manifest

akgrenSoar committed:
Merge pull request #1 from akgrenSoar/development

A-Enums

akgrenSoar committed:
Level-7

akgrenSoar committed:
Merge branch 'branch-Level-7'

akgrenSoar committed:
Merge Level-8

akgrenSoar committed:
A-MoreOOP

akgrenSoar committed:
Functional

akgrenSoar committed:
A-JUnit

akgrenSoar committed:
Minor Cleanup

akgrenSoar committed:
A-CodingStandard

akgrenSoar committed:
branch-Level-9

akgrenSoar committed:
Add task enum

akgrenSoar committed:
Liskov Substitution

akgrenSoar committed:
A-JavaDoc

akgrenSoar committed:
Merge branch 'branch-A-JavaDoc'

akgrenSoar committed:
Merge branch 'branch-A-CodingStandard'

akgrenSoar committed:
Merge branch 'branch-Level-9'

akgrenSoar committed:
Code Cleanup

akgrenSoar committed:
OOP

akgrenSoar committed:
Add Storage

akgrenSoar committed:
Add Parser

akgrenSoar committed:
Parser JUnit

akgrenSoar committed:
Minor IO Cleanup

akgrenSoar committed:
Minor Clean Import

akgrenSoar committed:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606'

akgrenSoar committed:
Gradle Support

akgrenSoar committed:
Add Comments and JavaDoc

akgrenSoar committed:
Add Gradle Custom Task

JingYenLoh commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,75 @@
+     */
+    public static void save(List<Task> taskList, String filePath) throws IOException {
+        // Open file for write/overwrite (throws IOException)
+        FileWriter fileWriter = new FileWriter(filePath);
```
Perhaps a try with block can be used here, for an auto flush and close?

akgrenSoar replied:
Thanks for pointing that out! I've included try-with-resources to ensure that fileWriter is automatically closed.

JingYenLoh commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,43 @@
+        System.out.println("Here are the matching tasks in your list:");
+
+        // Search matching tasks
+        AtomicInteger index = new AtomicInteger(0);
```
What's the reason for using an `AtomicInteger`? Perhaps a traditional for loop would have sufficed here?

akgrenSoar replied:
It seems like i got into the heat of practicing functional programming.
Like you pointed out, a normal for loop would've been the simpler option.
The change will be reflected in the next commit.

JingYenLoh commented on text-ui-test/runtest.sh:
```diff
@@ -13,7 +13,7 @@ then
 
 # compile the code into the bin folder, terminates if error occurred
-if ! javac -cp ../src -Xlint:none -d ../bin ../src/main/java/Duke.java
+if ! javac -cp ../src -Xlint:none -d ../bin ../src/main/java/*.java ../src/main/java/cmd/*.java ../src/main/java/command/*.java ../src/main/java/misc/*.java ../src/main/java/task/*.java
```
Perhaps globstar could be enabled, so that this script doesn't have to change every time a new package is added?

EDIT: I realize this is out of the scope of the review but ya

akgrenSoar replied:
Nice comment, I've learnt something new today!
I went ahead and attempted to adapt globstar in runtest.bat, but it seems like there are problems using ** and *.java together in batch script. Nevertheless, i will find time to implement and test it on runtest.sh. Thanks for the suggestion!

For reference, in runtest.bat, i tried
javac  -cp ..\src -Xlint:none -d ..\bin ..\src\main\java\duke\Main.java ..\src\main\java\duke\**\*.java // Illegal char '*'
javac  -cp ..\src -Xlint:none -d ..\bin ..\src\main\java\duke\**\*.java // Illegal char '*'
javac  -cp ..\src -Xlint:none -d ..\bin ..\src\main\java\duke\** // works, but dir only

Do let me know if i missed anything.

akgrenSoar replied:
Update: Globstar seems to work on shell script, but sunfire and compute clusters doesn't seem to have java11. Nevertheless, I've modified runtest.sh to use globstar.

JingYenLoh replied:
Globstar is a Bash feature, it does not exist in batch scripts. Sunfire is ancient and really just used for jumping. There might be several nodes with Java 11. Perhaps `xgpe1`?

akgrenSoar replied:
Yup, managed to verify on xgpe1. Tyvm

JingYenLoh reviewed (commented):
Other than the individual review comments, perhaps most of the Javadoc comments can follow the coding standard:

> the first sentence should start in the form Returns ..., Sends ..., Adds ...

but I shall let your other reviewer handle that :smile: 

Overall, LGTM. I'm not sure if it's "the most robust duke in CS2103" 😉, but it's definitely clean and readable, with well thought out exception handling!

akgrenSoar committed:
Adapt Peer Review - JingYenLoh

Peer review changes
 - Change stream to forloop (FindCommand.java)
 - Close fileWriter on Exception (Storage.java)
 - Add globstar (runtest.sh)
 - Fix problems (runtest.sh)

Others
 - Improve readability (Parser.java)

akgrenSoar committed:
Add DukeParserException

DukeParserException
 - Remove InvalidCommand.java in favour of DukeParserException
 - Reflect changes in
    - CommandFactory.java
    - Parser.java
    - ParserTest.java

Others
 - Change stream to forloop in HelpCommand.java
 - Change stream to forloop in ListCommand.java

akgrenSoar committed:
Fix Done Symbol

akgrenSoar commented:
> 
> 
> Other than the individual review comments, perhaps most of the Javadoc comments can follow the coding standard:
> 
> > the first sentence should start in the form Returns ..., Sends ..., Adds ...
> 
> but I shall let your other reviewer handle that 😄
> 
> Overall, LGTM. I'm not sure if it's "the most robust duke in CS2103" 😉, but it's definitely clean and readable, with well thought out exception handling!

Hi Jing Yen Loh,

Thanks the the review and the helpful tips. The individual comments you gave aligned with my principles, and i have included them in the next commit.

I am aware that there is room for improvements in the code, especially in unit testing and javadoc. But with each comment, it will get closer and closer to becoming the most robust duke in CS2103 😃

pengxiangg commented on src/main/java/duke/command/ReversibleCommand.java:
```diff
@@ -0,0 +1,13 @@
+/**
+ * ReversibleCommand interface implements execute() and reverse() methods
+ */
+public interface ReversibleCommand extends Command {
```
I like how you a ReversibleCommand to allow users to undo their commands.

pengxiangg commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,75 @@
+        Supplier<String> nextLine = () -> scanner.hasNext() ? scanner.nextLine() : null;
+
+        // Process each line as stream
+        Stream.generate(nextLine)
```
I like the use of streams

pengxiangg reviewed (commented):
LGTM, neat and tidy code

akgrenSoar committed:
Rename CommandFactory.java to ParseToCommand.java

akgrenSoar committed:
Initial GUI (Empty)

akgrenSoar committed:
Simple GUI

akgrenSoar committed:
Show user input in GUI

akgrenSoar committed:
Add launch via CMD or GUI
Add launch via CMD or GUI
Fix text-ui-test

akgrenSoar committed:
Add GUI ListView

akgrenSoar committed:
Fix usage message

akgrenSoar committed:
Add Undo/Redo feature

akgrenSoar committed:
Add Logic class

akgrenSoar committed:
Check Code Quality

akgrenSoar committed:
Improve Code Quality
Improve Code Quality in Storage.java

akgrenSoar committed:
Refractor ParseToCommand

akgrenSoar committed:
Add Assertions

akgrenSoar committed:
Merge pull request #2 from akgrenSoar/branch-A-Assertions

Add Assertions

akgrenSoar committed:
Merge branch 'master' into branch-A-CodeQuality

akgrenSoar committed:
Create gradle.yml

Automatically check commits & pull requests with github actions

akgrenSoar committed:
Merge branch 'master' of https://github.com/akgrenSoar/ip

akgrenSoar committed:
Update runtest.sh

akgrenSoar committed:
Fix JUnit Test on Unix and MacOS

akgrenSoar committed:
Add Assertions

akgrenSoar committed:
Duke A-BetterGui

akgrenSoar committed:
Merge branch 'branch-A-BetterGui'

akgrenSoar committed:
Fix Args Parsing

akgrenSoar committed:
Add CommandException

akgrenSoar committed:
Add Ui.png

akgrenSoar committed:
Fix Import Order

akgrenSoar committed:
Clean runtest

akgrenSoar committed:
Refractor duke.core

akgrenSoar committed:
Fix Assertion

akgrenSoar committed:
Add MainWindow (Partial FXML)

akgrenSoar committed:
Add FXML

akgrenSoar committed:
A-BetterGui

akgrenSoar committed:
Code Cleanup

akgrenSoar committed:
Add Logger

akgrenSoar committed:
User Guide

akgrenSoar committed:
Fix UserGuide Syntax

akgrenSoar committed:
Update project readme
